analyze_transformations.py

```diff
@@ -46,36 +32,80 @@ def easy_to_transform_definition(m: Macro,
     ])
 
 
-def main():
-    ap = argparse.ArgumentParser()
-    ap.add_argument('results_file', type=str)
-    ap.add_argument('-o', '--output_file')
-    args = ap.parse_args()
+def generate_macro_translations(mm: MacroMap) -> dict[Macro, str | None]:
+    translationMap: dict[Macro, str | None] = {}
+
+    for macro, invocations in mm.items():
+        # We only need to look at the first invocation to determine the translation
+        # Already verified that all type signatures are the same
+        invocation = next(iter(invocations), None)
+        assert invocation is not None
+
+        # Static to avoid breaking the one definition rule
+        if macro.IsFunctionLike:
+            # return only if not void
+            returnStatement = "return" if not invocation.TypeSignature.startswith("void") else ""
+            translationMap[macro] = f"static inline {invocation.TypeSignature} {{ {returnStatement} {macro.Body}; }}"
+
+        elif macro.IsObjectLike:
+            # All invocations where an ICE is required must be representable by type int 
+            # to be translatable to an enum
+            can_translate_to_enum = all(
+                [i.IsICERepresentableByInt32 for i in invocations if i.IsInvokedWhereICERequired])
+
+            # If no invocations require ICE, just make it a static const variable
+            not_invoked_where_ICE_required = all([not i.IsInvokedWhereICERequired for i in invocations])
+
+            if not_invoked_where_ICE_required:
+                translationMap[macro] = f"static const {invocation.TypeSignature} = {macro.Body};"
+            elif can_translate_to_enum:
+                translationMap[macro] = f"enum {{ {macro.Name} = {macro.Body} }};"
+            else:
+                translationMap[macro] = None
+
+    return translationMap
+
 
-    with open(args.results_file) as fp:
+def get_interface_equivalent_preprocessordata(results_file: str) -> PreprocessorData:
+    unique_names : dict[str, Any] = {}
+
+    with open(results_file) as fp:
         entries = json.load(fp)
 
+        # Keep only one macro definition. if there's more than one, keep none
+        for obj in entries:
+            if obj["Kind"] == "Definition":
+                if obj["Name"] in unique_names:
+                    unique_names[obj["Name"]] = None
+                else:
+                    unique_names[obj["Name"]] = obj
```

Comment:
This seems like a good place for a `Counter()`, but this approach should work as well.

Reply:
Actually this doesn't work right now and I didn't commit my changes to fix it, which did use Counter. Will commit them soon.